bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/Hors.java

```diff
@@ -0,0 +1,168 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liato.bankdroid.banking.banks;
+
+import android.content.Context;
+import android.text.InputType;
+
+import com.liato.bankdroid.Helpers;
+import com.liato.bankdroid.banking.Account;
+import com.liato.bankdroid.banking.Bank;
+import com.liato.bankdroid.banking.Transaction;
+import com.liato.bankdroid.banking.exceptions.BankChoiceException;
+import com.liato.bankdroid.banking.exceptions.BankException;
+import com.liato.bankdroid.banking.exceptions.LoginException;
+import com.liato.bankdroid.legacy.R;
+import com.liato.bankdroid.provider.IBankTypes;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.message.BasicNameValuePair;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import eu.nullbyte.android.urllib.Urllib;
+
+public class Hors extends Bank {
+
+    private static final String TAG = "Hors";
+
+    private static final String NAME = "Hörs";
+
+    private static final String NAME_SHORT = "hors";
+
+    private static final String URL = "http://www.dittkort.se/hors/";
+
+    private static final int BANKTYPE_ID = IBankTypes.HORS;
+
+    private static final boolean DISPLAY_DECIMALS = false;
+
+    private String response = null;
+
+    public Hors(Context context) {
+        super(context);
+        super.TAG = TAG;
+        super.NAME = NAME;
+        super.NAME_SHORT = NAME_SHORT;
+        super.BANKTYPE_ID = BANKTYPE_ID;
+        super.URL = URL;
+        super.DISPLAY_DECIMALS = DISPLAY_DECIMALS;
+        super.INPUT_TYPE_USERNAME = InputType.TYPE_CLASS_TEXT
+                | +InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS;
+        super.INPUT_HINT_USERNAME = context.getString(R.string.card_id);
+        super.INPUT_HIDDEN_PASSWORD = true;
+    }
+
+    public Hors(String username, String password, Context context) throws BankException,
+            LoginException, BankChoiceException, IOException {
+        this(context);
+        this.update(username, password);
+    }
+
+    @Override
+    protected LoginPackage preLogin() throws BankException, IOException {
+        urlopen = new Urllib(context);
+        urlopen.setAllowCircularRedirects(true);
+        response = urlopen.open("https://www.dittkort.se//q/?p=7EB4F129-0A41-417F-8FEA-51B2B75B9D24");
+
+        Document d = Jsoup.parse(response);
```

Comment:
You could break out L87-L100 to a new method named `private Document asDocument(String response)` to make it more readable.